client/v3/maintenance.go

```diff
@@ -78,6 +84,21 @@ type Maintenance interface {
 	Downgrade(ctx context.Context, action int32, version string) (*DowngradeResponse, error)
 }
 
+// SnapshotResponse is aggregated response from the snapshot stream.
+// Consumer is responsible for closing steam by calling .Snapshot.Close()
+type SnapshotResponse struct {
```

Comment:
I guess we cannot define this in the proto as it differs from it due to the `Snapshot`?

Reply:
Yes, we cannot put golang interface as part of proto. Having a struct will make the response extendable while keeping easy to consume ReaderCloser interface.